sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -310,13 +357,127 @@ def from_work(self, map):
         if self.tiled: return tilemap.redistribute(map, self.comm, self.rhs.geometry.active)
         else: return utils.allreduce(map, self.comm)
 
-    def write(self, prefix, tag, m):
+    def write(self, prefix, tag, m, write_partial_fits=False):
         if not self.output: return
         oname = self.ofmt.format(name=self.name)
         oname = "%s%s_%s.%s" % (prefix, oname, tag, self.ext)
-        if self.tiled:
-            tilemap.write_map(oname, m, self.comm)
+        if self.pix_scheme == "rectpix":
+            if self.tiled:
+                tilemap.write_map(oname, m, self.comm)
+            else:
+                if self.comm is None or self.comm.rank == 0:
+                    enmap.write_map(oname, m)
         else:
-            if self.comm is None or self.comm.rank == 0:
-                enmap.write_map(oname, m)
+            if self.ext in ["fits", "fits.gz"]:
+                if hp is None:
+                    raise ImportError("Cannot save healpix map as fits; healpy could not be imported. Install healpy or save as npy or h5")
+                if m.ndim > 2:
+                    m = np.reshape(m, (np.product(m.shape[:-1]), m.shape[-1]), order='C') # Flatten wrapping axes; healpy.write_map can't handle >2d array
+                hp.write_map(oname, m.view(self.dtype), nest=(self.hp_geom.ordering=='NEST'), partial=write_partial_fits, overwrite=True)
+            elif self.ext == "npy":
+                np.save(oname, m.view(self.dtype))
+            elif self.ext in ['h5', 'hdf5']:
+                if h5py is None:
+                    raise ValueError("Cannot save healpix map as hdf5; h5py could not be imported. Install h5py or save as npy or fits")
+                with h5py.File(oname, 'w') as f:
+                    dset = f.create_dataset("data", m.shape, dtype=self.dtype, data=m)
```

Comment:
Same here -- add attributes to this dataset with important projection info.

Reply:
Done, added ordering and nside as attributes